httpx/_types.py

```diff
@@ -41,6 +41,8 @@
     "Headers",
     Dict[str, str],
     Dict[bytes, bytes],
+    Dict[str, Optional[str]],
+    Dict[bytes, Optional[bytes]],
```

Comment:
Would this be enough? A `Dict[str, str]` is _also_ a `Dict[str, Optional[str]]` (as in "it's a sub-type").

```suggestion
    Dict[str, Optional[str]],
    Dict[bytes, Optional[bytes]],
```

Reply:
mypy complaints here due to invariance. More info: https://mypy.readthedocs.io/en/latest/common_issues.html#variance

We have to use explicit types in this case or create a custom type and then use it in replacement of `Dict[str, str]`

Reply:
…Or should we switch to `Mapping` instead of `Dict`? I believe it's covariant in value types.

Reply:
Switching to `Mapping` helps. I believe it's a small change.